ui/src/app/layout/topics/deletetopic/delete-topic.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Component, Input } from '@angular/core';
+import { Topic, TopicsService } from '../../../shared/services/topics.service';
+import { EnvironmentsService } from '../../../shared/services/environments.service';
+import { ServerInfoService } from '../../../shared/services/serverinfo.service';
+import { TranslateService } from '@ngx-translate/core';
+import { ToastService } from '../../../shared/modules/toast/toast.service';
+import { take } from 'rxjs/operators';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { Router } from '@angular/router';
+
+@Component({
+    selector: 'app-delete-topic-component',
+    templateUrl: './delete-topic.component.html'
+})
+export class DeleteTopicComponent {
+
+    @Input() topic: Topic;
+
+    @Input() isOwnerOfTopic: boolean;
+
+    @Input() topicName: string;
+
+    @Input() translateParams: any = {};
+
+    topicNameConfirmText = '';
+
+    constructor(
+        private serverInfoService: ServerInfoService,
```

Comment:
Please check constructor params. Not all services seem to be used in this class.